internal/addrs/module.go

```diff
@@ -170,3 +173,109 @@ func (m Module) Ancestors() []Module {
 func (m Module) configMoveableSigil() {
 	// ModuleInstance is moveable
 }
+func (m Module) configRemovableSigil() {}
+
+func parseModulePrefix(traversal hcl.Traversal) (Module, hcl.Traversal, tfdiags.Diagnostics) {
```

Comment:
Maybe I'm not well-versed enough in the codebase, but I don't know what a module prefix is and what the naming rules would be to parse them by. It may be useful to add a comment here explaining what the naming rules are we are parsing by.

Reply:
You are correct! I wanted to add an explanation here and forgot. I'll add it.